scripts/utils/smoke-test-webpack.mjs

```diff
@@ -0,0 +1,49 @@
+import { dirname, join } from 'path';
+import { fileURLToPath } from 'url';
+import assert from 'node:assert';
+
+if (process.argv.length !== 3) {
+  console.error('usage: node smoke-test-webpack.mjs <webpack-version>');
+  process.exit(1);
+}
+
+const webpack = (await import(`webpack-${process.argv[2]}`)).default;
+
+const root = dirname(fileURLToPath(import.meta.url));
+const output = {
+  path: join(root, 'tmp'),
+  filename: 'webpack.bundle.mjs',
+  library: {
+    type: 'module',
+  },
+  chunkFormat: 'module',
+};
+
+webpack(
+  {
+    mode: 'development',
+    entry: '../scripts/utils/smoke-test-webpack.import.mjs',
+    output,
+    target: 'node',
+    experiments: { outputModule: true },
+    devtool: false,
+  },
+  async (err, stats) => {
+    if (err !== null) {
+      console.error('compilation error', err);
+      process.exit(1);
+    }
+
+    console.log('stats', stats.toString());
+
+    if (err !== null || stats.hasErrors()) {
+      process.exit(1);
+    }
+
+    const { makeTpufRequest } = await import(join(output.path, output.filename));
+    const res = await makeTpufRequest();
+    assert(res.status === '🐡');
+
+    console.log(`webpack smoke test passed (webpack v${webpack.version})`);
+  },
```

Comment:
might be better to go with this (we're checking for err twice and not checking if stats is undefined):
```suggestion
  async (err, stats) => {
    if (err !== null || (stats && stats.hasErrors())) {
      console.error('compilation error', err);
      process.exit(1);
    }
    
    console.log('stats', stats.toString());

    const { makeTpufRequest } = await import(join(output.path, output.filename));
    const res = await makeTpufRequest();
    assert(res.status === '🐡');

    console.log(`webpack smoke test passed (webpack v${webpack.version})`);
  },
```

Reply:
Ah, whoops, thanks! I refactored that logic at some point when I realized how `err` and `stats` interact and didn't quite nail it. Pushed up a slightly different version of your proposal that should avoid the redundancy. (It's possible for `err === null && stats.hasErrors()`, which means you need to print the stats to show the error. webpack is weird.)